core/src/main/java/org/codice/imaging/nitf/core/image/impl/ImageSegmentParser.java

```diff
@@ -128,7 +136,7 @@ public final ImageSegmentImpl parse(final NitfReader nitfReader, final ParseStra
             readXBANDS();
         }
         for (int i = 0; i < numBands; ++i) {
-            ImageBand imageBand = new ImageBand();
+            ImageBandImpl imageBand = new ImageBandImpl();
```

Comment:
I updated this to return ImageBand, but I think there's a significant amount of refactoring that still needs to be done which is outside the scope of this one story. We have several ...Impl classes that are public and shouldn't be. They're often used only in tests and we can replace those with mock objects. Separating the interfaces is just one step toward securing the API, but we'll need other stories to really tighten the API up.